### ISSUE TITLE
Can i use oehost.lib in MT(multi thread) mode in windows?

### ISSUE THREAD

**Issue by Mojicode** [openenclave/openenclave#4208]
my application and it's dependency all compile in MT mode, and when i use office vs extention and nuget package, i found it is MD.
So i clone the openenclave and recompile it in MT,  after solved some error,  i compiled the openenclave in MT。
but when i use it in vs with replace auto generated file such like oehost.lib, it can not create enclave:

`Exception thrown at 0x000001C7ED16F94A (MyEnclave1.elf.signed) (located in ConsoleApplication1.exe): 0xC0000005: An access violation occurred while writing location 0xFFFF81C7ED39B000.`

so i guess the oehost can not be use in MT?

**Comment by radhikaj** [comment]
oehost.lib is a static lib.

**Comment by radhikaj** [comment]
@fchinchilla has this question - Are there different versions of compilers being used to build the code and a different version to build the lib?
@Mojicode  - How did you determine that it is compiled using MD?

**Comment by Mojicode** [comment]
> @fchinchilla has this question - Are there different versions of compilers being used to build the code and a different version to build the lib?
> @Mojicode - How did you determine that it is compiled using MD?

According to these infos below:
1. In windows, binary have to be compiled in the same mode with it's dependency, i set my binary to MT, when i link office oehost, there are some errors;
2. use the vs tool : dumpbin, the command is [dumpbin.exe  /directives xxx.lib], in the output content, the " /DEFAULTLIB" field show the type for MT or MD:

- MSVCRT means MD
- MSVCRTD means MDd
- LIBCMT means MT
- LIBCMTD means MTd
and the office oehost is MSVCRT, and the oehost i compiled is LIBCMT.

during my compilation, there was some errors, i fixed it locally, but i think these errors won't  have effect on the final lib.

**Comment by Mojicode** [comment]
Is there possible to support a openenclave with MT compile mode? especial for lib that host app use.

**Comment by anakrish** [comment]
@Mojicode I will take a look.

**Comment by Mojicode** [comment]
> @Mojicode I will take a look.

Hi, I have successfully compiled my application in MT mode using my local oehost.lib compiled in MT mode. The error before is caused by mixed use old enclave *.a lib and new oehost.lib I compiled. The oehost.lib was compile use the master branch code that supposed to be 0.17.1 .  But the enclave *.a lib was release by the nuget package which was 0.17.0 at that moment(the new package has updated to 0.17.1).

But I encounter to another crash, i will open a new issue, and close this one. Thank you.
